File: app/src/main/java/es/etologic/mahjongscoring2/injection/AppModule.java
package es.etologic.mahjongscoring2.injection;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
class AppModule {

    @Provides
    @Singleton
    Context provideContext(MahjongScoringApp mahjongScoringApp) { return mahjongScoringApp; }
}

File: app/src/main/java/es/etologic/mahjongscoring2/injection/AppModule.kt
package es.etologic.mahjongscoring2.injection

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
internal class AppModule {
    
    @Provides
    @Singleton
    internal fun provideContext(mahjongScoringApp: MahjongScoringApp): Context = mahjongScoringApp
}
